src/Tasks/GetCompatiblePlatform.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Build.Tasks
+{
+    class GetCompatiblePlatform : TaskExtension
+    {
+        public ITaskItem[] AnnotatedProjects { get; set; }
+
+        public string ParentProjectPlatform { get; set; }
+
+        public string PlatformLookupTable { get; set; }
+
+        [Output]
+        public ITaskItem[] AssignedProjectsWithPlatform { get; set; }
+
+        public override bool Execute()
+        {
+            Dictionary<string, string> translationTable = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            if (!string.IsNullOrEmpty(PlatformLookupTable))
+            {
+                foreach (string s in PlatformLookupTable.Split(';'))
+                {
+                    // Minimum translation: a=b
+                    if (s.Length < 3)
+                    {
+                        continue;
+                    }
+                    string key = s.Split('=')[0];
+                    string val = s.Split('=')[1];
+                    translationTable[key] = val;
+                }
+                Log.LogMessage($"Translation Table: {translationTable.Aggregate(new StringBuilder(), (sb, kvp) => sb.Append(kvp.Key + "=" + kvp.Value + ";"), sb => sb.ToString())}");
+            }
+
+            AssignedProjectsWithPlatform = new ITaskItem[AnnotatedProjects.Length];
+            for (int i = 0; i < AnnotatedProjects.Length; i++)
+            {
+                AssignedProjectsWithPlatform[i] = new TaskItem(AnnotatedProjects[i]);
```

Comment:
I'll save that for a future update to this feature. Reasons:

1. We can always improve perf after we get this in
2. The `GetReferenceNearestTargetFrameworkTask` does a similar copy
	- Source for that is in Nuget.Build.Tasks.dll
3. Trying to get this merge sooner rather than later